CI-CD_DOCUMENTATION.md

```diff
@@ -0,0 +1,286 @@
+# Notepads CI/CD documentation
+
+* after merging the PR, the first run of the "Notepads CI/CD Pipeline" workflow will not complete successfully, because it requires specific setup explained in this documentation. The two other workflows "CodeQL Analysis" and "Build", should complete successfully.
+
+## 1. Set up SonarCloud
+### SonarCloud is a cloud-based code quality and security service
+
+#### Create SonarCloud project
+
+- Go to https://sonarcloud.io/
+
+- Click the "Log in" button and create a new account or connect with GitHub account (recommended)
+
+- At the top right corner click the "+" sign
+
+- From the dropdown select "Create new Organization"
+
+- Click the "Choose an organization on GitHub" button
+
+- Select an account for the organization setup
+
+- On Repository Access select "Only select repositories" and select the project and click the "Save" button
+
+- On the "Create organization page" don't change the Key and click "Continue"
+
+- Select the Free plan then click the "Create Organization" button to finalize the creation of the Organization
+
+#### Configure SonarCloud project
+
+- At the top right corner click the "+" sign and select "Analyze new project"
+
+- Select the project and click the "Set Up" button in the box on the right
+
+- Under "Choose your analysis method" click "With GitHub Actions" and **keep the following page open**
+
+- [Create a new PAT with **repo_deployment** and **read:packages** permissions](#6-how-to-create-a-pat) and copy the value of the generated token
+
+- In the project's GitHub repository, go to the **Settings** tab -> Secrets 
+
+- Click on **New Repository secret** and create a new secret with the name **SONAR_GITHUB_TOKEN** and the token you just copied as the value
+
+- Create another secret with the two values from the SonarCloud page you kept open, which you can close after completing this step
+
+![SonarCloud_1](/ScreenShots/CI-CD_DOCUMENTATION/SonarCloud_1.png)
```

Comment:
Keeping `/` in the beginning of path doesn't retender image preview both in VSCode and Visual Studio. Does the editor you use renders images in preview?? I would suggest to change it to  `ScreenShots/CI-CD_DOCUMENTATION/SonarCloud_1.png` or `./ScreenShots/CI-CD_DOCUMENTATION/SonarCloud_1.png`